src/_js/map.js

```diff
@@ -200,24 +200,48 @@
 		});
 
 		const mapContainer = document.querySelector('#map');
-		const MARKERS_URL_PATH = mapContainer.dataset.markerPack ? mapContainer.dataset.markerPack : 'markers.json';
-		const xhr = new XMLHttpRequest();
-		xhr.open('GET', URL_PREFIX + MARKERS_URL_PATH);
-		xhr.responseType = 'json';
-		xhr.onload = function() {
-			if (xhr.status === 200) {
-				xhr.response.forEach(m => {
-					const options = {'title': m.description};
-					if (m.icon && m.icon in icons) { options.icon = icons[m.icon]; }
-					if (!_this.markersLayers[m.z]) { _this.markersLayers[m.z] = new L.layerGroup(); }
-					_this.markersLayers[m.z].addLayer(
-						L.marker(_this.map.unproject([m.x + 0.5, m.y + 0.5], 0), options)
-					);
-				});
-				_this._tryShowMarkers();
-			}
-		};
-		xhr.send();
+		const urlParams = new URLSearchParams(window.location.search);
+		// Possible markers sources
+		// A) https://example.com?m=<base64-json-str>#32368,32198,7:0
+		// B) https://example.com?mf=https://example.com/pack.json#32368,32198,7:0
+		// C) <div id="map" data-marker-json="<json-str>" ...>
+		// D) <div id="map" data-marker-pack="https://example.com/pack.json" ...>
```

Comment:
Hmm, now that we have all these options, I'd like to change the old name for clarity. Let's change this to `data-markers` (for the new JSON one) and `data-markers-url` (for the old one), respectively.